src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -650,6 +627,97 @@ int IEqualityComparer<ImmutableHashSet<string>>.GetHashCode(ImmutableHashSet<str
             return hash;
         }
 
+        private class DisplayNameToItemsMap : IEnumerable<CompletionItem>
```

Comment:
considre not directly exposing this as a IEnumerable<CompletionItem>.  will prevent anything from going through teh interface-dispatch pathways here.

Reply:
What do you mean by this? I'm passing this to ArrayBuilder.AddRange(), so I think the interface is required?

Reply:
ah ok.  another option would be for you to enumerate this yourself, calling ArrayBuilder.Add.

But if this is only one allocation, it's nbd.